components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -121,12 +127,13 @@ type rtkSerial struct {
 
 	activeBackgroundWorkers sync.WaitGroup
 
-	mu            sync.Mutex
-	ntripMu       sync.Mutex
-	ntripconfigMu sync.Mutex
-	ntripClient   *rtk.NtripInfo
-	ntripStatus   bool
-	isClosed      bool
+	mu               sync.Mutex
+	ntripMu          sync.Mutex
+	ntripconfigMu    sync.Mutex
+	urlMutex         sync.Mutex
```

Comment:
I get nervous when a struct has 2 mutexes, and now we've got 4. How do you ensure we can't deadlock? If you're relying on locking them in a specific order, please describe it in a comment so future developers don't accidentally introduce a deadlock.